tempesta_fw/ss_skb.c

```diff
@@ -656,13 +729,18 @@ __skb_fragment(struct sk_buff *skb, struct sk_buff *pskb,
 		if ((offset >= 0) && (offset < d_size)) {
 			int t_size = d_size - offset;
 			len = max(len, -t_size);
-			ret = __split_pgfrag(skb, i, offset, len, it);
+			ret = __split_pgfrag(skb, bskb, i, offset, len, it);
 			goto done;
 		}
 	}
 
+	/* skb fragments are not expected in skb fragments. */
+	if (skb != bskb)
+		return -ENOENT;
```

Comment:
Shouldn't we use `WARN_ON()` instead? This isn't an input error, but actually our assumption about how underlying code works. And this assumption can be correct or not. So we should see errors or warnings rather than dropping client connections.